plasmapy/formulary/frequencies.py

```diff
@@ -519,3 +520,56 @@ def upper_hybrid_frequency(B: u.T, n_e: u.m**-3) -> u.rad / u.s:
 
 wuh_ = upper_hybrid_frequency
 """Alias to `~plasmapy.formulary.frequencies.upper_hybrid_frequency`."""
+
+
+@validate_quantities(
+    omega_p={"can_be_negative": False},
+    validations_on_return={
+        "units": [u.m**-3],
+        "equivalencies": [
+            (u.F * u.kg * u.rad**2 / (u.C**2 * u.m * u.s**2), u.m**-3)
+        ],
+    },
+)
+def plasma_critical_density(omega_p: u.rad / u.s) -> u.m**-3:
+    r"""Calculate the plasma critical density.
+
+    Parameters
+    ----------
+    omega_p : `~astropy.units.Quantity`
+        The plasma frequency in radians per second.
+
+    Returns
+    -------
+    n_c : `~astropy.units.Quantity`
+        The plasma critical density.
+
+    Raises
+    ------
+    `TypeError`
+        If ``omega_p`` is not a `~astropy.units.Quantity`.
+
+    `~astropy.units.UnitConversionError`
+        If ``omega_p`` is not in correct units.
+
+    `ValueError`
+        If ``omega_p`` is an invalid value.
+
+    Warns
```

Comment:
Is this true for your code, or left over from another docstring?

Reply:
Yep this is correct, a negative frequency will raise a ValueError

Reply:
@namurphy just opened a new PR to address this in the guidelines (#1667 ): the docstring should only document errors that are explicitly raised by the function itself, not the validator. So, you should remove these errors + warnings here.